components/switch/switch-mixin.js

```diff
@@ -252,8 +252,20 @@ export const SwitchMixin = superclass => class extends FocusMixin(RtlMixin(super
 
 	_toggleState() {
 		if (this.disabled) return;
+
+		const beforeChangeEvent = new CustomEvent('d2l-switch-before-change', {
+			detail: {
+				update: on => this.on = on
```

Comment:
Should the `change` event fire here too? Otherwise I don't think it's ever going to fire if the `before-change` stuff is being used.

Reply:
Ah yes, thanks. I missed that when I was mirroring `d2l-button-toggle`. It actually waits till `updated` to dispatch its `d2l-button-toggle-change` event. My thinking is that we should similarly dispatch after Lit has updated.